[AIRFLOW-6983] Use SingletonThreadPool for database communication

Khrol opened the pull request:
Tasks are executed in CLI mode when connection pool to the database is disabled.
`settings.configure_orm(disable_connection_pool=True)`

While one task is run, multiple DB communications are happening while a separate connection is allocated for each of them. It results in DB failures.

Having SingletonThreadPool allows us to reuse the same connection. Also SingletonThreadPool is useful for other places (WebServer, Scheduler) because they are single-threaded.

---
Issue link: [AIRFLOW-6983](https://issues.apache.org/jira/browse/AIRFLOW-6983)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] ~Unit tests coverage for changes (not needed for documentation changes)~ Not sure if this is relevant for such configuration udpates
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

ashb reviewed (changes requested):
This is 100% against the comment that is right before it. Please dig in to which is correct.

You've copied the comment into the config description, but reading that no one would ever set it to anything but the default True

Khrol committed:
[AIRFLOW-6983] Use SingletonThreadPool for database communication

SingletonThreadPool a better fit for the single-threaded, single process engine like Airflow.

ashb commented on airflow/settings.py:
```diff
@@ -148,18 +148,6 @@ def configure_orm(disable_connection_pool=False):
         # 0 means no limit, which could lead to exceeding the Database connection limit.
         pool_size = conf.getint('core', 'SQL_ALCHEMY_POOL_SIZE', fallback=5)
```
Shouldn't we remove this too?

Khrol replied:
https://docs.sqlalchemy.org/en/13/core/pooling.html#sqlalchemy.pool.SingletonThreadPool

`SingletonThreadPool` has `pool_size` parameter.
